src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -344,6 +348,55 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             }
         }
 
+        /// <summary>
+        /// Checks whether unused targets exist. If found any, just write an internal log at Warn level.
+        /// <remarks>If initializing not started or failed, then checking process will be canceled</remarks>
+        /// </summary>
+        private void CheckUnusedTargets()
+        {
+            if (this.InitializeSucceeded == null)
+            {
+                InternalLogger.Warn("Unused target checking is canceled -> initialize not started yet.");
+                return;
+            }
+            if (this.InitializeSucceeded.Value == false)
+            {
+                InternalLogger.Warn("Unused target checking is canceled -> initialize not succeeded.");
+                return;
+            }
+
+            var swatch = Stopwatch.StartNew();
+
+            ReadOnlyCollection<Target> configuredNamedTargets = this.ConfiguredNamedTargets; //assign to variable. because it is returning new ReadOnlyCollection everytime.
+            InternalLogger.Warn("Unused target checking is started... Rule Count: {0}, Target Count: {1}", this.LoggingRules.Count, configuredNamedTargets.Count);
+
+            HashSet<string> targetNamesAtRules = new HashSet<string>();
+            for (int ri = 0; ri < this.LoggingRules.Count; ri++)
```

Comment:
No, I am not confused with linq-to-sql.

To be sure whether Linq will broke something, I searched which version has support. Maybe my conclusion is wrong. But I could not found any result for supporting older than .Net 3.5 ([without extra effort] (http://stackoverflow.com/questions/3348348/is-there-a-way-to-use-linq-query-syntax-with-net-3-0-projects))
[At least Linq MSDN page has not older version] (https://msdn.microsoft.com/en-us/library/system.linq(v=vs.90).aspx).